### PR TITLE
GatherTree description was extended and outdated link fixed

### PR DESCRIPTION
**Pull request by sadolini**
Description: GatherTree description was extended and outdated link fixed

JIRA: 38239

Code: **NA doc changed**
* [ ]  Comments 
* [ ]  Code style (PEP8)
* [ ]  Transformation generates reshape-able IR
* [ ]  Transformation preserves original framework node names

Validation: **NA doc changed**
* [ ]  Unit tests
* [ ]  Framework operation tests
* [ ]  Transformation tests
* [ ]  e2e model test with an update of ./tests/e2e_oss/utils/reshape_utils.py
* [ ]  Model Optimizer IR Reader check

Documentation: **NA existing  doc was fixed**
* [ ]  Supported frameworks operations list
* [ ]  Supported **public** models list
* [ ]  New operations specification
* [ ]  Guide on how to convert the **public** model
* [ ]  User guide update

### CODE REVIEW

**Review by maxnick — COMMENTED** [review]
Regarding the pseudocode, please check the Jira ticket. I've added a version of the pseudocode that was implemented as a reference.

**Comment by lazarevevgeny** [comment]
@sadolini , once this PR is reviewed and approved, please, create a PR for the release branch 'releases/2021/1'

**Review by maxnick — APPROVED** [review]

**Review by lazarevevgeny — APPROVED** [review]

File: docs/ops/movement/GatherTree_1.md [hunk 1]
```diff
@@ -8,21 +8,34 @@
 
 **Detailed description**
 
-GatherTree operation implements the same algorithm as GatherTree operation in TensorFlow. Please see complete documentation [here](https://www.tensorflow.org/versions/r1.12/api_docs/python/tf/contrib/seq2seq/gather_tree?hl=en).
+GatherTree operation implements the same algorithm as GatherTree operation in TensorFlow.
+Please see complete documentation [here](https://www.tensorflow.org/addons/api_docs/python/tfa/seq2seq/gather_tree).
```
Comment (aalborov) [inline docs/ops/movement/GatherTree_1.md:12]: --> `The GatherTree operation implements the same algorithm as the [GatherTree operation in TensorFlow](https://www.tensorflow.org/addons/api_docs/python/tfa/seq2seq/gather_tree).`

**Review by aalborov — CHANGES_REQUESTED** [review]

File: docs/ops/movement/GatherTree_1.md (hunk 1 above)
Comment (sadolini) [reply docs/ops/movement/GatherTree_1.md]: fixed

**Review by aalborov — APPROVED** [review]

**Review by lazarevevgeny — APPROVED** [review]

**Pull request merged**